### ISSUE TITLE
Env: Error in `readWordPressVersion`

### ISSUE THREAD

**Issue by swissspidy** [WordPress/gutenberg#43222]
### Description

It looks like either #41780 or #42826 broke wp-env usage for the Preferred Languages feature plugin.

Trying to update to `@wordpress/env` 5.1.0 (which includes this change) breaks `wp-env start` with this error:

```

✖ Cannot read property 'path' of null
TypeError: Cannot read property 'path' of null
    at readWordPressVersion (/home/runner/work/preferred-languages/preferred-languages/node_modules/@wordpress/env/lib/wordpress.js:251:14)
    at start (/home/runner/work/preferred-languages/preferred-languages/node_modules/@wordpress/env/lib/commands/start.js:138:4)
```

### Step-by-step reproduction instructions

Using a wp-env config file like this:

```json
{
	"core": null,
	"plugins": [ "." ]
}
```

Try to run `wp-env start`

### Environment info

`@wordpress/env` 5.1.0

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
I ran into the same problem.
I believe it is due to the fact that #42826 no longer allows `null` in the `core` property.

https://github.com/WordPress/gutenberg/pull/42826/files#r933608144

I consider this to be a breaking change and believe that either of the following actions should be taken: 

- Specify in the changelog that it is a breaking change. In addition, properly update [the README of the package](https://github.com/WordPress/gutenberg/blob/trunk/packages/env/README.md#wp-envjson)
- Add fallback to allow `null` values

It is also strange that [phpVersion allows null](https://github.com/WordPress/gutenberg/blob/6dcf85a1261adf1e3e2b676986644fc520386884/packages/env/lib/config/config.js#L103) while core does not.

Personally, I think both properties should accept null.

@noahtallen
@gziolo 

I'd like to hear your thoughts 🙂

**Comment by noahtallen** [comment]
Actually already have a fix here: #43133

**Comment by noahtallen** [comment]
The fix will allow `null` again -- it was just an oversight on my part that it didn't work in the first place.

**Comment by noahtallen** [comment]
By the way, for what it's worth, `"core": null,` has no effect by itself in a plain `.wp-env.json`. The absence of the `core` property means the same thing, which is to use the stable version. However, it is useful in `.wp-env.override.json` if you want to "clear" the version the environment has and instead use whatever the latest stable version is.

**Comment by noahtallen** [comment]
The fix should now be available in version 5.1.1 :)
